lock/lock.go

```diff
@@ -108,19 +108,26 @@ func (d *Driver) ExecuteWithMultiUseLock(ucs []config.UseLocker, timeout time.Du
 // lock after a timeout, returning a stop channel. Timeout is jittered to
 // mitigate thundering herd problems.
 func (d *Driver) AcquireWithTTLRefresh(uc config.UseLocker, ttl, timeout time.Duration) (chan struct{}, error) {
+	if err := d.Config.PublishUse(uc); err != nil {
+		return nil, err
+	}
+
 	if err := d.acquire(uc, ttl, timeout); err != nil {
 		return nil, err
 	}
 
-	stopChan := make(chan struct{})
+	stopChan := make(chan struct{}, 1)
```

Comment:
TTL refresh would hang in a number of spots otherwise while the sleep 
was happening.

On 22 Aug 2016, at 14:04, Bill Robinson wrote:

>>  	if err := d.acquire(uc, ttl, timeout); err != nil {
>>  		return nil, err
>>  	}
>>
>> -	stopChan := make(chan struct{})
>> +	stopChan := make(chan struct{}, 1)
>
> I understand the implication of this change (the first send on the 
> channel will not block now), but what was the reason for the change?
>
> -- 
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/contiv/volplugin/pull/435/files/e1d8c9c912325bd3cc9499e2277ec812df753329..06f347455c320e9fa60e8ff924edd5000b7fe6cb#r75758459